nbextensions/bulkImportCell/main.js

```diff
@@ -0,0 +1,157 @@
+     */
+    function setupNotebook() {
+        return Promise.all(Jupyter.notebook.get_cells().map(function(cell) {
+            if (BulkImportCell.isBulkImportCell(cell)) {
+                try {
+                    const bulkImportCell = new BulkImportCell(cell);
+                }
+                catch(error) {
+
+                    // If we have an error here, there is a serious problem setting up the cell and it is not usable.
+                    // What to do? The safest thing to do is inform the user, and then strip out the cell, leaving
+                    // in it's place a markdown cell with the error info.
+                    // For now, just pop up an error dialog;
+                    var ui = UI.make({
+                        node: document.body
+                    });
+                    ui.showInfoDialog({
+                        title: 'Error',
+                        body: div({
+                            style: {
+                                margin: '10px'
+                            }
+                        }, [
+                            ui.buildPanel({
+                                title: 'Error Starting App Cell',
+                                type: 'danger',
+                                body: ui.buildErrorTabs({
+                                    preamble: p('There was an error starting the app cell.'),
+                                    error: error
+                                })
+                            })
+                        ])
+                    });
+                }
+            }
+        }));
+    }
+
+    /*
+     * Called directly by Jupyter during the notebook startup process.
+     * Called after the notebook is loaded and the dom structure is created.
+     * The job of this call is to mutate the notebook and cells to suite
+     * oneself, set up any services or other things needed for operation of
+     * the notebook or cells.
+     * The work is carried out asynchronously through an orphan promise.
+     */
+    function load_ipython_extension() {
+
+        setupNotebook()
+            .then(() => {
+                $([Jupyter.events]).on('insertedAtIndex.Cell', (event, payload) => {
+                    const cell = payload.cell,
+                        setupData = payload.data,
+                        jupyterCellType = payload.type;
+
+                    if (jupyterCellType !== 'code' ||
+                        !setupData ||
+                        !(setupData.type === CELL_TYPE)) {
+                        return;
+                    }
+
+                    try {
+                        const bulkImportCell = new BulkImportCell(cell, true);
+                    }
+                    catch(error) {
+                        console.error('Error while creating bulk import cell', error);
+                        Jupyter.notebook.delete_cell(Jupyter.notebook.find_cell_index(cell));
+                        var ui = UI.make({
```

Comment:
almost the same code as above... time to make a general function? `reportCellError(cellType, errorType)`?

Reply:
Yup, and move it to a common module, and (eventually) refactor all cell extensions to use it.

Reply:
done. I moved it to `common/error.js` which doesn't really do the same things, but seemed appropriate.